mne_bids/read.py

```diff
@@ -415,17 +429,30 @@ def read_raw_bids(bids_fname, bids_root, extra_params=None,
             coordsystem_json = json.load(fin)
 
         # Get coordinate frames that electrode coordinates are in
+        # Note: all coordinate frames should be lower-case
         if kind == "meg":
-            coord_frame = coordsystem_json['MEGCoordinateSystem']
+            coord_frame = coordsystem_json['MEGCoordinateSystem'].lower()
+            coord_unit = coordsystem_json['MEGCoordinateUnits']
         elif kind == "ieeg":
-            coord_frame = coordsystem_json['iEEGCoordinateSystem']
-        else:
+            coord_frame = coordsystem_json['iEEGCoordinateSystem'].lower()
+            coord_unit = coordsystem_json['iEEGCoordinateUnits']
+
+            # default coordinate frames to available ones in mne-python
+            # noqa: see https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html
+            if coord_frame not in _ieeg_coordinate_frames_dict:
+                warn("Defaulting coordinate frame to MRI "
+                     "from coordinate system input {}".format(coord_frame))
+                coord_frame = 'mri'
+        else:  # noqa
             raise RuntimeError("Kind {} not supported yet for "
                                "coordsystem.json and "
                                "electrodes.tsv.".format(kind))
-        # read in electrode coordinates and attach to raw
-        raw = _handle_electrodes_reading(electrodes_fname, coord_frame, raw,
-                                         verbose)
+
+        if coord_frame is not None:
```

Comment:
what about MEG? Should you pass this down also for MEG data?

Reply:
MEG doesn't have an electrodes.tsv 

See: https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/02-magnetoencephalography.html